roles/neutron-data-network/tasks/main.yml

```diff
@@ -43,142 +43,61 @@
-    owner: root
-    group: neutron
-    mode: 0644
-  notify: restart neutron lbaas agent
-  when: (neutron.lbaas.enabled == "smart" and
-         groups['controller'][0] not in groups['compute']) or
-         neutron.lbaas.enabled|bool
-
-- name: symlink other neutron lbaas haproxy templates
-  file:
-    src: /opt/openstack/current/neutron/lib/python2.7/site-packages/neutron_lbaas/drivers/haproxy/templates/{{ item }}
-    dest: /etc/neutron/lbaas_templates/{{ item }}
-    owner: root
-    group: neutron
-    state: link
-    force: true
-  with_items:
-    - haproxy.loadbalancer.j2
-    - haproxy_proxies.j2
-  notify: restart neutron lbaas agent
-  when:
-    - openstack_install_method != 'distro'
-    - (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool
-
-- name: symlink other neutron lbaas haproxy templates (distro)
-  file:
-    src: /usr/lib/python2.7/site-packages/neutron_lbaas/drivers/haproxy/templates/{{ item }}
-    dest: /etc/neutron/lbaas_templates/{{ item }}
-    owner: root
-    group: neutron
-    state: link
-    force: true
+- name: install neutron-data-network services (rhel)
+  systemd_service:
+    name: "{{ item.name }}"
+    description: "{{ item.desc }}"
+    type: "{{ item.type }}"
+    user: "{{ item.user }}"
+    environments: "{{ neutron.service.envs }}"
+    cmd: "{{ item.cmd }}"
+    config_dirs: "{{ item.config_dirs }}"
+    config_files: "{{ item.config_files }}"
+    pidfile: "{{ item.pidfile|default(omit) }}"
+    restart: "{{ item.restart }}"
+    kill_mode: "{{ item.kill_mode }}"
   with_items:
-    - haproxy.loadbalancer.j2
-    - haproxy_proxies.j2
-  notify: restart neutron lbaas agent
-  when:
-    - openstack_install_method == 'distro'
-    - (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool
-
-- name: neutron lbaas agent config
-  template:
-    src: etc/neutron/services/loadbalancer/haproxy/lbaas_agent.ini
-    dest: /etc/neutron/services/loadbalancer/haproxy
-    mode: 0644
-  when: (neutron.lbaas.enabled == "smart" and
-         groups['controller'][0] not in groups['compute']) or
-         neutron.lbaas.enabled|bool
-  notify:
-    - restart neutron lbaas agent
+    - "{{ neutron.services.neutron_dhcp_agent }}"
+    - "{{ neutron.services.neutron_l3_agent }}"
+    - "{{ neutron.services.neutron_metadata_agent }}"
+  when: ursula_os == 'rhel'
+
+- block:
+  - include: lbaas-haproxy-cleanup.yml
+    when: neutron.lbaas.f5.enabled | bool
+
+  - include: lbaas-f5-cleanup.yml
+    when: neutron.lbaas.f5.enabled | bool == false
+
+  - include: lbaas-f5.yml
+    when: neutron.lbaas.f5.enabled | bool
+
+  - include: lbaas-haproxy.yml
+    when: neutron.lbaas.f5.enabled | bool == false
+  when: (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool
+
+- block:
```

Comment:
the previous "when condition" locates in the block with when so that we won't be able to be update the "when condition" for disabling all lbaasv2.
~~
  when: (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool
~~~

I tried to consolidate all together instead of including the same yml file for multiple times. It looks concise without blocks and probably with better performance, but with less readability. 

~~~
- include: lbaas-haproxy-cleanup.yml
  when: not (neutron.lbaas.enabled|bool or neutron.lbaas.f5.enabled|bool or (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']))

- include: lbaas-f5-cleanup.yml
  when: not neutron.lbaas.f5.enabled|bool
    
- include: lbaas-haproxy.yml
  when: 
      - (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool)
      - not neutron.lbaas.f5.enabled|bool

- include: lbaas-f5.yml
  when: 
      - (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool)
      - neutron.lbaas.f5.enabled|bool 
~~~

leave the existing code for comparison.
~~~
- block:
  - include: lbaas-haproxy-cleanup.yml
    when: neutron.lbaas.f5.enabled | bool

  - include: lbaas-f5-cleanup.yml
    when: not neutron.lbaas.f5.enabled | bool

  - include: lbaas-f5.yml
    when: neutron.lbaas.f5.enabled | bool

  - include: lbaas-haproxy.yml
    when: not neutron.lbaas.f5.enabled | bool
  when: (neutron.lbaas.enabled == "smart" and groups['controller'][0] not in groups['compute']) or neutron.lbaas.enabled|bool

- block:
  - include: lbaas-f5-cleanup.yml
  - include: lbaas-haproxy-cleanup.yml
  when:  not neutron.lbaas.enabled|bool
~~~